clarify aws bedrock confusion (#9357)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-ai/wizard-byok.md

# Configure BYOK for <Constant name="wizard" /> <Lifecycle status="beta"/>

<IntroText>

> The <Constant name="wizard" /> CLI runs in bring-your-own-key (BYOK) mode, which means you supply credentials from a supported provider.

Reviewer: ```suggestion
You can use the <Constant name="wizard" /> CLI with bring-your-own-key (BYOK), which means you supply credentials from a supported provider.
```

</IntroText>

<WizardFeedbackCallout />

> The BYOK instructions on this page apply to the CLI only. The <Constant name="dbt_platform" /> uses a separate [account-level integration](/docs/platform/enable-dbt-ai). BYOK works as follows:

Reviewer: Hiya Mirna

This sentence slightly conflicts with the Supported AI providers table on the same page: That table covers both dbt platform and CLI. It might be worthwhile rewording (e.g. “The setup steps below are for the CLI...)

Author: updated to 'on this page' and didn't add 'below' since we dont use directional words

- <Constant name="wizard" /> calls your chosen provider's API directly using your key.
- Usage costs appear on your provider account.
- Token costs are billed by whichever provider you choose.

<WizardSupportedProviders />

## Configure a provider

You can configure a provider in one of the following ways:

- **Terminal commands**: Best for users who want to configure providers from the shell.

> - **Interactive setup**: Best for most users working in the <Constant name="wizard" /> TUI. Use the `/providers` slash command.

Reviewer: ```suggestion
- **Interactive session**: Best for most users working in the <Constant name="wizard" /> TUI. Use the `/providers` slash command.
```

- **Environment variables**: Best for headless runs, such as `wizard exec`, automation, or temporary local sessions.

### Configure in the terminal

In the terminal, use the `providers` subcommand to list, configure, and enable providers:

```bash
wizard providers list
wizard providers configure PROVIDER_NAME
wizard providers enable PROVIDER_NAME
```

> The `configure` command prompts you to enter credentials for the selected provider. To use environment variables or set a key without echoing it in your shell history, refer to [Set your API key](#set-your-api-key).

Reviewer: Hiya Mirna

Does configure bedrock prompt for credentials, or is Bedrock env var + enable + set-region the intended flow? If the latter, maybe add a one-line Bedrock exception or link to the Bedrock section

Author: it prompts  for credentials.

For example:

```bash
wizard providers configure openai

> wizard providers enable openai

Reviewer: Hiya Mirna

This example only shows `openai` but the same page documents Bedrock (and Snowflake) with different steps. A reader setting up Bedrock may copy the OpenAI pattern. Maybe add a note that provider setup differs (e.g. “For AWS Bedrock, refer to [AWS Bedrock](#aws-bedrock)”)...

Author: i think havingthe 'for example' probably solves this

  6. azure                disabled; 3/3 models selected; missing credentials; needs setup
  7. snowflake            disabled; 3/3 models selected; missing credentials; needs setup
  8. gemini               disabled; 3/3 models selected; missing credentials; needs setup
```

## Set your API key

<Tabs>
<TabItem value="interactive" label="Interactive session" default>

The first time you start <Constant name="wizard" /> in a project, onboarding prompts you to choose a provider.

1. At the **Configure a Provider** prompt, select your provider.
2. Paste your API key or provider credentials when prompted.
3. Choose an AI model to finish setup.

To add or switch providers later from an active session, type `/providers` in the TUI.

When you configure a provider through <Constant name="wizard" />, credentials are stored in `~/.dbt/wizard/provider-auth.json`.

</TabItem>
<TabItem value="env-var" label="Environment variable">

Set the key as an environment variable if you want to:

* Run <Constant name="wizard" /> in headless mode, such as with `wizard exec`.
* Use a key for the current terminal session only.
* Reuse the same key across sessions.

> * Avoid storing credentials in the <Constant name="wizard" /> config directory.

Reviewer: Hiya Mirna

Bullets 2 and 3 can read as conflicting (session-only vs across sessions). Maybe merge the two or clarifying how each works, e.g. “Use a key for one session (export in the terminal) or across sessions (add to your shell profile).”

export ANTHROPIC_API_KEY="sk-ant-..."
```

</TabItem>
<TabItem value="bedrock" label="Amazon Bedrock">

```bash
export AWS_BEARER_TOKEN_BEDROCK="ABSK..."
```

</TabItem>
</Tabs>

To make an environment variable available across terminal sessions, add it to your shell profile, such as `.zshrc`, `.bashrc`, or equivalent.

> Avoid committing API keys to dotfiles, project files, or shared configuration.

Reviewer: Line 1 recommends adding the export to `.zshrc/.bashrc`, but line 2 warns against committing keys to dotfiles. It might be worth rewording line 2 to “Avoid committing API keys to version control, project files, or shared configuration.”
